src/main/java/net/dries007/tfc/objects/items/wood/ItemBucketTFC.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Work under Copyright. Licensed under the EUPL.
+ * See the project README.md and LICENSE.txt for more information.
+ */
+
+package net.dries007.tfc.objects.items.wood;
+
+import java.util.HashSet;
+import java.util.Set;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.annotation.ParametersAreNonnullByDefault;
+
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.init.SoundEvents;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.ActionResult;
+import net.minecraft.util.EnumActionResult;
+import net.minecraft.util.EnumHand;
+import net.minecraft.util.SoundCategory;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.RayTraceResult;
+import net.minecraft.world.World;
+import net.minecraftforge.common.capabilities.ICapabilityProvider;
+import net.minecraftforge.fluids.Fluid;
+import net.minecraftforge.fluids.FluidRegistry;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.FluidUtil;
+import net.minecraftforge.fluids.capability.CapabilityFluidHandler;
+import net.minecraftforge.fluids.capability.IFluidHandler;
+
+import mcp.MethodsReturnNonnullByDefault;
+import net.dries007.tfc.ConfigTFC;
+import net.dries007.tfc.api.capability.FluidWhitelistCapability;
+import net.dries007.tfc.api.capability.size.Size;
+import net.dries007.tfc.api.capability.size.Weight;
+import net.dries007.tfc.objects.items.ItemTFC;
+
+@MethodsReturnNonnullByDefault
+@ParametersAreNonnullByDefault
+public class ItemBucketTFC extends ItemTFC
+{
+    public static final Set<Fluid> ALLOWED_BUCKET_FLUIDS;
+    private static final int CAPACITY = 1000;
+
+    static
+    {
+        ALLOWED_BUCKET_FLUIDS = new HashSet<>();
+        for (String fluidName : ConfigTFC.GENERAL.woodenBucketWhitelist)
+        {
+            ALLOWED_BUCKET_FLUIDS.add(FluidRegistry.getFluid(fluidName));
```

Comment:
I have one concern here: this could potentially run too early, before fluids have been registered. TFC will register Fluids -> Blocks -> Items, so it shouldn't be a problem there, but other mods may wait until init or after items.

My suggestion would be to move this to an on-demand / lazy loaded calculation in `initCapabilities`, as I doubt it's that resource intensive.